index.d.ts

```diff
@@ -122,13 +122,16 @@ export interface ResponsePromise extends Promise<Response> {
  * The error has a response property with the `Response` object.
  */
 export class HTTPError extends Error {
+	constructor(response: Response);
 	response: Response;
 }
 
 /**
  * The error thrown when the request times out.
  */
-export class TimeoutError extends Error {}
+export class TimeoutError extends Error {
+	constructor();
```

Comment:
Can TypeScript not figure this out on its own? Seems silly.

Reply:
It was using the Error constructor which accepts a message.

Reply:
Yeah, I get the intention of the type definition. I'm just a bit surprised that TypeScript can't figure out that this function doesn't accept any arguments.

Reply:
I believe that Typescript would be smart enough to infer the proper constructor if our implementation was written in Typescript.

Reply:
Yeah, perhaps. Seems like a good feature request, in any case.